src/url/url.service.ts

```diff
@@ -0,0 +1,14 @@
+import { Injectable, Inject } from '@nestjs/common';
+import { REQUEST } from '@nestjs/core';
+import { Request } from 'express';
+
+@Injectable()
+export class UrlService {
+  constructor(@Inject(REQUEST) private readonly request: Request) {}
+
+  getFullUrl(path: string): string {
+    const protocol = this.request.protocol;
+    const host = this.request.get('host');
+    return `${protocol}://${host}${path}`;
```

Comment:
**suggestion**: Validate `protocol` and `host` to ensure they are not undefined or null before constructing the URL.